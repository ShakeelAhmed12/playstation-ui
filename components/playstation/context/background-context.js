import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const BackgroundProvider = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}