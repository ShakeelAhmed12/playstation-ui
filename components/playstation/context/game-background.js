import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const GameBackground = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}