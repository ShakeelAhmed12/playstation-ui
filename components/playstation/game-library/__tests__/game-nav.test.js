import React from 'react';
import renderer from 'react-test-renderer';
import { GameBackground } from '../../context/game-background';

import GameNav from '../game-nav';

it('Game nav renders correctly', () => {
    const tree = renderer
        .create(
            <GameBackground>
                <GameNav />
            </GameBackground>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});