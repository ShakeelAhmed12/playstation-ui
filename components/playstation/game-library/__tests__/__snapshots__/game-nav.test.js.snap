// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="game-nav__StyleGameNav-r40yoh-0 idRwUJ"
>
  <div
    className="game active"
    onClick={[Function]}
  >
    <svg
      height="56"
      viewBox="0 0 24 24"
      width="56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M21 3L3 10.53v.98l6.84 2.65L12.48 21h.98L21 3z"
        fill="white"
      />
    </svg>
  </div>
  <div
    className="game "
  >
    <div
      className="game-item__GameImage-sc-11p7q2j-0 dNqegf"
      onClick={[Function]}
    />
  </div>
  <div
    className="game "
  >
    <div
      className="game-item__GameImage-sc-11p7q2j-0 gAIaqg"
      onClick={[Function]}
    />
  </div>
  <div
    className="game "
  >
    <div
      className="game-item__GameImage-sc-11p7q2j-0 gRCxuN"
      onClick={[Function]}
    />
  </div>
  <div
    className="game "
  >
    <div
      className="game-item__GameImage-sc-11p7q2j-0 eoAzkO"
      onClick={[Function]}
    />
  </div>
  <div
    className="game "
  >
    <div
      className="game-item__GameImage-sc-11p7q2j-0 hGUJoE"
      onClick={[Function]}
    />
  </div>
  <div
    className="game "
  >
    <div
      className="game-item__GameImage-sc-11p7q2j-0 ZeGRG"
      onClick={[Function]}
    />
  </div>
</section>
`;
